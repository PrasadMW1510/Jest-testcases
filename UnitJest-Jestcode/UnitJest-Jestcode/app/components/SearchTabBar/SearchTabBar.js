import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import Tab from '../Tab/index';
import './SearchTabBar.scss';

class SearchTabBar extends React.Component {
  render() {
    const props = {
      activeClassName: 'tab-bar__tab--active',
      component: NavLink,
    };
    return (
      <div className="tab-bar search-tab-bar">
        <div className="tab-bar__container search-tab-container">
          <Tab {...props} className="tab-bar__tab--adv-search" to="/books/quiz/advanced">
            Advanced Search
          </Tab>
          <Tab
            {...props}
            className="tab-bar__tab--results"
            to="/books/quiz/results"
            onClick={this.props.checkSearch}
          >
            Search Results
          </Tab>
          <Tab {...props} className="tab-bar__tab--custom" to="/books/quiz/custom">
            Custom List
          </Tab>
        </div>
      </div>
    );
  }
}

SearchTabBar.propTypes = {
  checkSearch: PropTypes.func,
};

export default SearchTabBar;
