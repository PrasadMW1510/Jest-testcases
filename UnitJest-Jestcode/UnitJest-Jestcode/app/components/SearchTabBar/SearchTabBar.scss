.search-tab-wrapper {
  margin: 10px;
}

.search-tabs-content {
  border: 2px solid #000;
  background-color: #fff;
  height: 402px;
  margin-top: -2px;

  p {
    font-family: inherit;
  }
}

.search-tab-bar {
  background-color: transparent;
}

.search-tab-container {
  left: 0;

  .tab-bar__tab {
    font-size: 13px;
  }

  .tab-bar__text {
    margin-left: 8px;
    padding-left: 0;
  }

  .tab-bar__tab::before {
    background-color: #ff9e61;
  }

  .tab-bar__tab--active::before {
    background-color: #fff;
  }
}
