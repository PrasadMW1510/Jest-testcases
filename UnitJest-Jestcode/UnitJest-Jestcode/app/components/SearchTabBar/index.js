/**
 *
 * SearchTabBar
 *
 */

export { default } from './SearchTabBar';
