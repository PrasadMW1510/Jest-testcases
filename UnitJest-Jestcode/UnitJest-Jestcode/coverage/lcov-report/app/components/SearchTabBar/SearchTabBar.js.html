<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/SearchTabBar/SearchTabBar.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">app/components/SearchTabBar</a> SearchTabBar.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/51</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React <span class="cstat-no" title="statement not covered" >from 'react';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
import PropTypes <span class="cstat-no" title="statement not covered" >from 'prop-types';<span class="cstat-no" title="statement not covered" ></span></span>
import { NavLink } fro<span class="cstat-no" title="statement not covered" >m 'react-router-dom';</span>
import Tab fr<span class="cstat-no" title="statement not covered" >om '../Tab/index';<span class="cstat-no" title="statement not covered" ></span></span>
<span class="cstat-no" title="statement not covered" >import './SearchTabBar.scss';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
class SearchTab<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >Ba</span>r extends React.Component {<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span>
  render() {
    const props = <span class="cstat-no" title="statement not covered" >{</span>
      activeClassName: 'tab-bar__tab--active',
      component: NavLink,
    };
    re<span class="cstat-no" title="statement not covered" >turn (</span>
      &lt;div className="tab-bar search-tab-bar"&gt;
        &lt;div className="tab-bar__container search-tab-container"&gt;
          &lt;Tab {...props} className="tab-bar__tab--adv-search" to="/books/quiz/advanced"&gt;
            Advanced Search
          &lt;/Tab&gt;
          &lt;Tab
            {...props}
            className="tab-bar__tab--results"
            to="/books/quiz/results"
            onClick={this.props.checkSearch}
          &gt;
            Search Results
          &lt;/Tab&gt;
          &lt;Tab {...props} className="tab-bar__tab--custom" to="/books/quiz/custom"&gt;
            Custom List
          &lt;/Tab&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }<span class="cstat-no" title="statement not covered" ></span>
}
&nbsp;
<span class="cstat-no" title="statement not covered" >SearchTabBar.propTypes = {</span>
  checkSearch: PropTypes.func,<span class="cstat-no" title="statement not covered" ></span>
};
&nbsp;
export default SearchTabBar;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 21 2018 17:06:47 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
