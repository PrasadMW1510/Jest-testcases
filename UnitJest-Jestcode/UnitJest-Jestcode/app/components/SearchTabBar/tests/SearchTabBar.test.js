import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchTabBar from '../index';

describe('<SearchTabBar />', () => {
  let wrapper = null;
  let mockItemHandler = null;
  let mocklinkHandler = null;

  beforeEach(() => {
    mockItemHandler = jest.fn();
    mocklinkHandler = true;

    wrapper = shallow(<SearchTabBar checkSearch={mockItemHandler} setLink={mocklinkHandler} />);
  });

  it('Expect to render tab 1 correctly', () => {
    const bookTabBar1 = wrapper.find('.tab-bar__tab--adv-search');
    expect(bookTabBar1.exists()).toBe(true);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect to render tab 2 correctly', () => {
    const bookTabBar2 = wrapper.find('.tab-bar__tab--results');
    expect(bookTabBar2.exists()).toBe(true);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Expect to render tab 3 correctly', () => {
    const bookTabBar2 = wrapper.find('.tab-bar__tab--custom');
    expect(bookTabBar2.exists()).toBe(true);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
