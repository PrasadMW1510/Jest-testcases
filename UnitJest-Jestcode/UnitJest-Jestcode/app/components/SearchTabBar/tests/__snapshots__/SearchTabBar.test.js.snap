// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchTabBar /> Expect to render tab 1 correctly 1`] = `
<div
  className="tab-bar search-tab-bar"
>
  <div
    className="tab-bar__container search-tab-container"
  >
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--adv-search"
      component={[Function]}
      to="/books/quiz/advanced"
    >
      Advanced Search
    </Tab>
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--results"
      component={[Function]}
      onClick={[Function]}
      to="/books/quiz/results"
    >
      Search Results
    </Tab>
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--custom"
      component={[Function]}
      to="/books/quiz/custom"
    >
      Custom List
    </Tab>
  </div>
</div>
`;

exports[`<SearchTabBar /> Expect to render tab 2 correctly 1`] = `
<div
  className="tab-bar search-tab-bar"
>
  <div
    className="tab-bar__container search-tab-container"
  >
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--adv-search"
      component={[Function]}
      to="/books/quiz/advanced"
    >
      Advanced Search
    </Tab>
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--results"
      component={[Function]}
      onClick={[Function]}
      to="/books/quiz/results"
    >
      Search Results
    </Tab>
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--custom"
      component={[Function]}
      to="/books/quiz/custom"
    >
      Custom List
    </Tab>
  </div>
</div>
`;

exports[`<SearchTabBar /> Expect to render tab 3 correctly 1`] = `
<div
  className="tab-bar search-tab-bar"
>
  <div
    className="tab-bar__container search-tab-container"
  >
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--adv-search"
      component={[Function]}
      to="/books/quiz/advanced"
    >
      Advanced Search
    </Tab>
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--results"
      component={[Function]}
      onClick={[Function]}
      to="/books/quiz/results"
    >
      Search Results
    </Tab>
    <Tab
      activeClassName="tab-bar__tab--active"
      className="tab-bar__tab--custom"
      component={[Function]}
      to="/books/quiz/custom"
    >
      Custom List
    </Tab>
  </div>
</div>
`;
